packages/ai/core/util/chat-store.test.ts

```diff
@@ -0,0 +1,1090 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import type { UIMessage } from '../types';
+import { type ChatState, ChatStore } from './chat-store';
+
+const initStore = (chats?: Record<string, Pick<ChatState, 'messages'>>) => {
+  const store = new ChatStore({ chats });
+  return store;
+};
+
+describe('ChatStore', () => {
+  describe('initialization', () => {
+    it('initializes with a single chat', () => {
+      const id = 'chat-1';
+      const messages = [
+        { id: '1', content: 'hello', role: 'user', parts: [] },
+        { id: '2', content: 'world', role: 'assistant', parts: [] },
+      ] as UIMessage[];
+      const chats = {
+        [id]: { messages },
+      };
+      const store = initStore(chats);
+      expect(store.getMessages(id)).toEqual(messages);
+      expect(store.totalChats).toEqual(1);
```

Comment:
recommend `toMatchInlineSnapshot`

Reply:
your call! fyi in `v5` we plan to switch to inline snapshots more heavily (and there is `test:update` task for updating them)

Reply:
also curious what your concerns are (if updating is solved)?